chain/chain/src/store/mod.rs

```diff
@@ -2010,10 +2010,10 @@ impl<'a> ChainStoreUpdate<'a> {
         shard_id: ShardId,
         partial_storage: Option<PartialStorage>,
         applied_receipts_hash: CryptoHash,
-        contract_accesses: BTreeSet<CodeHash>,
-        contract_deploys: BTreeSet<CodeHash>,
+        contract_updates: ContractUpdates,
     ) {
         if let Some(partial_storage) = partial_storage {
+            let ContractUpdates { contract_accesses, contract_deploys } = contract_updates;
```

Comment:
This is the interaction layer with store, verifying whether it's okay to not have contract_accesses and contract_deploys sorted?

Reply:
Just checked, I think we are storing vec here, so should be fine

Reply:
We do not rely on sorted-ness of the hashes (eg. not keeping a hash of the list etc.) so using HashSet should be fine. We store the hashes in the DB as Vector but the storage is temporary and again does not depend on the sortedness.

Reply:
yeah two considerations, first I started using vector all over the place in the first PR and did not want to change the protocol and DB structures. Second, I thought vec would store them more compactly and we turn them into hashset when processing anyways, thus did not want to deal with a migration.